apps/back-end/controllers/record-controller.js

```diff
@@ -0,0 +1,195 @@
+const { Prisma, PrismaClient } = require('@prisma/client')
+
+const prisma = new PrismaClient({ log: ['query'] })
+
+const recordController = {
+  // create a new record
+  postReocrd: async (req, res, next) => {
+    try {
+      // #swagger.tags = ['Expense Record']
+      const { title, amount, note, categoryId } = req.body
+      // uncommend after adding authentication process
+      // const userId = req.user.id
+
+      if (!title || !amount || !categoryId)
+        return res.json(
+          'missing title or amount or category id to create a new expense record'
+        )
+
+      await prisma.record.create({
+        data: {
+          title,
+          amount,
+          note,
+          categoryId,
+          userId: 13,
+        },
+      })
+
+      res.json('successfully create a new record')
+    } catch (error) {
+      next(error)
+    }
+  },
+
+  // Get all expense records and accept query string
+  getExpenseRecords: async (req, res, next) => {
+    try {
+      // #swagger.tags = ['Expense Record']
+
+      // uncommend after adding authentication process
+      // const userId = req.user.id
+      // remove above after adding authentication process
+      const userIds = await prisma.$queryRaw`SELECT id FROM "User";`
+      const userId = userIds.map(({ id }) => id)[0]
+
+      // Get categoryIds
+      const result =
+        await prisma.$queryRaw`SELECT id FROM "Category" WHERE NOT "mainCategory" = 'Income' AND NOT "mainCategory" = 'Savings';`
```

Comment:
The exclusive query (`WHERE **NOT** "mainCategory"`) might work in the current sample with fewer categories.
It'd be better to use explicit syntax to query expense records.

Reply:
I'm thinking about separating our one category model into three models, income, expense and saving model. In the category model, I named the main category `Income`, but other expense categorise did not belong to any main category called `expense`. This design would make some difficulties while querying data.